### ISSUE TITLE
Feature Request: Add GCSFS to driver set for artifact tracking

### ISSUE THREAD

**Issue by ljstrnadiii** [neptune-ai/neptune-client#1084]
### Is your feature request related to a problem? Please describe.
Setting up gcs with s3 doesn't seem trivial and potentially limiting? Now we have to create a boto config for each user and all machines that will run neptune code, which might affect the way the underlying gsutil operates.

### Describe the solution you'd like
-  I'd like a lower barrier of entry to track files on gcs. We use [gcsfs](https://gcsfs.readthedocs.io/en/latest/), which is built on top of [fsspec](https://github.com/fsspec). 
- Another option is using google storage api directly.
- A non-s3 approach to tracking files on gcs or better documentation on how to setup s3 for gcs.
- Or maybe letting users register their own subclass of ArtifactDriver so you are off the hook?

### Describe alternatives you've considered
- creating a boto file for gcs, but I believe it might affect the underlying behavior of gcs and will also require another secret to pass around in remote settings...

**Comment by ljstrnadiii** [comment]
This is what w&b is doing: https://docs.wandb.ai/guides/artifacts/track-external-files#amazon-s3-gcs-references

**Comment by Blaizzy** [comment]
Hey @ljstrnadiii 

Prince here,

Could you tell me what version of the neptune-client you are running?

**Comment by ljstrnadiii** [comment]
Hey @Blaizzy! Thanks for the quick response. I am using `'0.16.9'

**Comment by Blaizzy** [comment]
We have introduced support for other S3-compatible storage providers from `v0.16.10` release of the neptune-client.

Please update your current version. You can use the following command :

`pip instal -U neptune-client`

Afterward, you can set up your gcs artifact tracking by passing the endpoint URL to an environment variable named `S3_ENDPOINT_URL`.

Example CLI command:
`export S3_ENDPOINT_URL='https://storage.googleapis.com'`

Example Python script:
```python
import os
os.environ["S3_ENDPOINT_URL"]='https://storage.googleapis.com'
...
```


Docs:
https://docs.neptune.ai/logging/artifacts/#configuration-for-other-providers

**Comment by Blaizzy** [comment]
Let me know if this works for you

**Comment by Blaizzy** [comment]
Please replace `https://storage.googleapis.com` with your gcs endpoint URL and try again.

`export S3_ENDPOINT_URL='YOUR_ENDPOINT_URL'`

**Comment by ljstrnadiii** [comment]
Does that mean I have to specify the bucket? Which endpoint here do I specify? https://cloud.google.com/storage/docs/request-endpoints

**Comment by ljstrnadiii** [comment]
Do I still use s3 protocol? Or do I use the gs protocol in specifying the url to the asset?

e.g. `gs://path/to/asset` or `s3://path/to/asset`?

**Comment by Blaizzy** [comment]
Hey @ljstrnadiii  

Sorry for the late response,

> Do I still use s3 protocol? Or do I use the gs protocol in specifying the url to the asset?

Yes, we still use the s3 protocol.

`s3://path/to/asset`  ✅

**Comment by ljstrnadiii** [comment]
No worries! Thanks for the help. I did try that, but am not sure what to specify for the S#_ENDPOINT_URL.

Do you mind helping me determine the `export S3_ENDPOINT_URL='YOUR_ENDPOINT_URL'` bit? I am not sure what to use here.

**Comment by Blaizzy** [comment]
> Do you mind helping me determine the export S3_ENDPOINT_URL='YOUR_ENDPOINT_URL

Yes, of course.

If you want to use GCS you can pass the following URL 'https://storage.googleapis.com'.
Example:
`export S3_ENDPOINT_URL='https://storage.googleapis.com'`

I just noticed I overlooked a small detail in the error message you sent.

You have to set the GCS credentials. For that, you can use the `AWS_SECRET_ACCESS_KEY` and `AWS_ACCESS_KEY_ID` environment variables.

Don't be mistaken by AWS_* prefix, those variables should be filled with [Google credentials](https://stackoverflow.com/a/60542236/17829528).

```
export AWS_SECRET_ACCESS_KEY='Your_GCS_key_here'
export AWS_ACCESS_KEY_ID='Your_GCS_ID_here'
```

**Comment by Blaizzy** [comment]
After the steps above, you can go ahead and start tracking artifacts in your GCS buckets

```
import neptune.new as neptune

run = init_run()

# Single file
run["train_dataset"].track_files("s3://datasets/images/image_1.png")
```
or

```
# Folder
run["train/images"].track_files("s3://datasets/images")
```

**Comment by Blaizzy** [comment]
Let me know if this works for you.

**Comment by ljstrnadiii** [comment]
Great, thanks for the clarification. Followup question: gcs often uses a env var to specify where a secret json lives that has these keys: 

```['client_x509_cert_url', 'auth_uri', 'private_key', 'client_email', 'private_key_id', 'client_id', 'token_uri', 'project_id', 'type', 'auth_provider_x509_cert_url']```

Any idea which elements map to `AWS_SECRET_ACCESS_KEY` and `AWS_ACCESS_KEY_ID`? I tried mapping both`private_key_id` and `client_id` to `AWS_ACCESS_KEY_ID` and `private_key` to `AWS_SECRET_ACCESS_KEY`, but no success.

Specifically, I get this

```botocore.exceptions.ClientError: An error occurred (SignatureDoesNotMatch) when calling the ListObjects operation: The request signature we calculated does not match the signature you provided. Check your Google secret key and signing method.```

**Comment by ljstrnadiii** [comment]
Success! I am not sure if there is a way around generating an HMAC key from the interoperability settings from the target bucket. 

TL;DR:
1. go to gcs bucket -> settings -> interoperability
2. map storage uri to `S3_ENDPOINT_URL`
3. generate a Service account HMAC `Access Key` maps to `AWS_ACCESS_KEY_ID` and `Secret Key` maps to `AWS_SECRET_ACCESS_KEY `

**Comment by SiddhantSadangi** [comment]
Hi @amabinsta 👋 

I appreciate you bringing this issue to our attention and sharing your idea for the feature. I agree that it would be useful to have the option to pass credentials to `track_files()` and use environment variables as a backup.

To help us prioritize and implement this feature, could you please create a new issue for this? It would only take a few minutes, and it would really help us with proper planning and tracking. I will make sure to forward it to the product team as soon as possible. 🙏
